### PR TITLE
Add initial module logic

### PR DESCRIPTION
**Pull request by vbudhram**
This is the initial module for hapi hpkp. It is based on the helmetjs version.

### CODE REVIEW

**Comment by jbuck** [comment]
Rather than hooking into onPreResponse as the integration step, would it be possible/desired to write the external interface as a hapi plugin? http://hapijs.com/tutorials/plugins

**Comment by vbudhram** [comment]
@jbuck Oh cool, didn't know you could do plugins. I will see if I can fit this in as one. Thanks!

**Comment by vbudhram** [comment]
Hey guys, finally had a chance to update this module. Added all things needed for it to be an Hapi plugin.

@jbuck @rfk @vladikoff Mind an r?

File: tests/index.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    it(testCase.name, function (done) {
+      server.inject(requestOptions, function (response) {
+        assert.equal(response.headers[testCase.expectedKey], testCase.expectedHeader)
+        done()
+      })
+    })
+  })
+})
+
+var inputTestCases = [
```
Comment (philbooth) [inline tests/index.js]: What is the meaning of `input` in this context? I think the tests that follow might read clearer if this was something like `invalidTestCases` or `failingTestCases` instead, unless that is inaccurate for some reason.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
+
+var hpkpOptions = {
+    maxAge: 1,                       // In seconds
+    sha256s: ["orlando=", "magic="], // Array of sha256
+    includeSubdomains: true,         // optional
+    reportUri: 'http://test.site',   // optional
+    reportOnly: false,               // optional
+}
+
+var server = new Hapi.Server();
```
Comment (philbooth) [inline README.md]: I know it's only example code, but should we be consistent about semi-colons? Either all with or all without I reckon.

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,4 @@
+extends: fxa/server
+
+rules:
+  handle-callback-err: 0
```
Comment (philbooth) [inline .eslintrc]: Do you actually need this rule? The only place I see an error to handle is the callback for `server.register` in `tests/index.js`, and it handles the error by calling `console.error`. Am I missing another?

**Comment by philbooth** [comment]
(whoops, only just noticed I was assigned-then-unassigned, feel free to ignore any of the above, obv)

**Comment by vbudhram** [comment]
Thanks @philbooth! Thought you were on vacation, didn't want to bother you :)

**Pull request merged**